### DISCUSSION TITLE
Model multiple Agents

### DISCUSSION THREAD

**Discussion by Erlix322** [JoelForamitti/agentpy#20] [Q&A] [answered]
Hey  Joël,

is there any example modeling multiple Agents of different types within a model?
I looked through the docs but I think I'm still missing something. :D 

Would I create different classes of the Agent type and create agent1, agent2,..., agent n in the Model to let them interact?

Maybe you can give me a quick overview on how to do this. 

BR
Eric

**Comment by jofmi** [comment, marked as answer]
Hi @Erlix322, you are right that there is no example for this in the docs yet. You would have to define two agent classes (e.g. `AgentType1` and `AgentType2`) and initiate a model for example as follows:

```python
class MyModel(ap.Model):
    def setup(self):
        self.type1_agents = ap.AgentList(self, 10, AgentType1)
        self.type2_agents = ap.AgentList(self, 10, AgentType2)
        self.all_agents = self.type1_agents + self.type2_agents
```
You can then use these three lists to access either agents of just one type, or all agents together.

**Reply by Erlix322** [reply]
Awesome! Thank you very much!
